go/vt/vttablet/grpctmclient/client.go

```diff
@@ -185,7 +186,19 @@ func (client *grpcClient) dialPool(ctx context.Context, tablet *topodatapb.Table
 
 	result := <-c
 	c <- result
-	return result.client, nil
+	invalidator := func(err error) {
+		if err == nil {
+			return
+		}
+		if _, ok := status.FromError(err); !ok {
+			// Not a gRPC error
+			return
+		}
```

Comment:
@harshit-gangal Would some error codes from MySQL also pass this check and end up causing us to invalidate the grpc connection?